commons/src/main/java/com/ozonehis/eip/security/oauth2/Oauth2Processor.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2021, Ozone HIS <info@ozone-his.com>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.ozonehis.eip.security.oauth2;
+
+import static com.ozonehis.eip.security.Constants.BEARER_HTTP_AUTH_SCHEME;
+import static com.ozonehis.eip.security.Constants.HEADER_OAUTH2_CLIENT_ID;
+import static com.ozonehis.eip.security.Constants.HEADER_OAUTH2_CLIENT_SECRET;
+import static com.ozonehis.eip.security.Constants.HEADER_OAUTH2_SCOPE;
+import static com.ozonehis.eip.security.Constants.HEADER_OAUTH2_URL;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.ProducerTemplate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Setter
+@Getter
+@Component("eip.oauthProcessor")
+public class Oauth2Processor implements Processor {
+
+    private final ProducerTemplate producerTemplate;
+
+    @Autowired
```

Comment:
Do we need this annotation here?

Reply:
Yes, we need it to inject `producerTemplate` Spring bean.